### ISSUE TITLE
Whitespace between elements is lost

### ISSUE THREAD

**Issue by yngwi** [nashwaan/xml-js#29]
Hi, thanks for your great library!

_xml-js version is v1.4.1_

It seems that whitespace between two sibling elements is lost when converted if it is the only content. The following code illustrates the problem:

```js
const convert = require('xml-js');
const xml = '<outer> This is <inner> some</inner> <inner>Text </inner> </outer>';
const json = convert.xml2json(xml, {compact: false, alwaysChildren: false});
const result = convert.json2xml(json, {compact: false});
console.log(result);
// <outer> This is <inner> some</inner><inner>Text </inner></outer>
```

As you can see, " This is " still has the whitespace after conversion, as well as " some " and " Text ", but not the whitespace between the first closing and second opening "inner" tags.

Is there some setting that I missed or is this a problem with the library?

Kind regards,
Daniel

**Comment by nashwaan** [comment]
Hi Daniel,

Thanks for reporting this issue and providing clear and concise example.

This behavior is by intention. Consider the following case where xml is not in one line:
```xml
<a>
     hi
    <b/>
     there
</a>
````
The spaces before `hi`, `<b/>`, and `there` are usually not required to be captured; otherwise the output will be unnecessary bloated.

However, if you still need to capture the white spaces between the elements as in your legitimate example, I have published an updated version of this library `v1.4.2` and supported your case under a hidden flag `captureSpacesBetweenElements`. Try this (after updating `xml-js` package):

```js
const convert = require('xml-js');
const xml = '<outer> This is <inner> some</inner> <inner>Text </inner> </outer>';
const json = convert.xml2json(xml, {captureSpacesBetweenElements: true});
const result = convert.json2xml(json);
console.log(result);
```

**Comment by yngwi** [comment]
Thank you very much, this is very convenient for us. Great support!
